go-controller/pkg/node/management-port_linux.go

```diff
@@ -311,7 +320,29 @@ func setupManagementPortNFTables(cfg *managementPortConfig) error {
 		),
 	})
 
+	isPodNetworkAdvertised := cfg.isPodNetworkAdvertised.Load()
+
 	if cfg.ipv4 != nil {
+		if isPodNetworkAdvertised {
+			tx.Add(&knftables.Rule{
+				Chain: nftablesMgmtPortChain,
+				Rule: knftables.Concat(
+					"meta nfproto ipv4",
+					"fib saddr type != local",
+					counterIfDebug,
+					"return",
+				),
+			})
+			tx.Add(&knftables.Rule{
+				Chain: nftablesMgmtPortChain,
+				Rule: knftables.Concat(
+					"meta nfproto ipv4",
+					"ip saddr", cfg.ipv4.ifAddr.IP,
```

Comment:
why is this necessary? isn't this the mpx IP?

Reply:
So if the source IP is already mpx IP, then there is no need to SNAT, is it? You can see more detail here:
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4572#issuecomment-2551052901

I guess this should not be specific to the advertised if block. Probably misplaced it when I rebased the nft changes.

Reply:
> I guess this should not be specific to the advertised if block. Probably misplaced it when I rebased the nft changes.

Changed accordingly, let me know if you would rather not.